utils/training_utils.py

```diff
@@ -86,14 +108,25 @@ def __iter__(self):
             self.batches = utils.batch_by_size(indices, self.dataset.num_tokens, max_tokens=self.max_tokens, max_sentences=self.max_sentences)
         else:
             self.batches = [indices[i:i + self.max_sentences] for i in range(0, len(indices), self.max_sentences)]
-            
-        for batch in self.batches:
+        
+        if self.shuffle:
+            rng.shuffle(self.batches)
+        
+        # TODO: correctly handle the indices outside of samplers
```

Comment:
Lightning seems to have problems dealing with leftover batches, which messes up their step count, resulting in irregular checkpoint model names. Temporarily randomly drop batches to make each epoch's batch count divisible by grad accumulation. Won't affect anyone with no grad accumulation.